### PR TITLE
Make Bonsai SDK Client Asynchronous

### PR DESCRIPTION
**Pull request by jjtny1**
Currently, the Bonsai SDK using `reqwest::blocking::Client` to make requests to Bonsai Alpha. However, we cannot use `reqwest::blocking` with `tokio` (https://github.com/seanmonstar/reqwest/issues/1233). This causes a panic when using the relayer (which uses the SDK to make calls to Bonsai) because the relayer relies heavily on `tokio` for asynchronous processing of proof requests.

This PR refactors the Client in the SDK to be asynchronous

### CODE REVIEW

File: bonsai/sdk/src/alpha.rs [hunk 1]
```diff
@@ -110,30 +110,30 @@ impl SessionId {
...
-            let body = res.text()?;
+            let body = res.text().await?;
             return Err(SdkErr::InternalServerErr(body));
         }
-        Ok(res.json::<SessionStatusRes>()?)
+        Ok(res.json::<SessionStatusRes>().await?)
     }
 }
 
 /// Represents a client of the REST api
```
Comment (capossele) [inline bonsai/sdk/src/alpha.rs:125]:
```suggestion
/// Represents a client of the REST api
#[derive(Debug, Clone)]
```

**Review by capossele — APPROVED** [review]
Let's wait for @mothran before merging though

File: bonsai/sdk/src/alpha.rs (hunk 1 above)
Comment (capossele) [reply bonsai/sdk/src/alpha.rs:125]: Ah and also we should make adjustments to the README to update the examples

File: bonsai/sdk/src/sync/mod.rs [hunk 2]
```diff
@@ -0,0 +1,485 @@
+// Copyright 2023 RISC Zero, Inc.
```
Comment (mothran) [inline bonsai/sdk/src/sync/mod.rs]: This file seems like a duplication of the async impl. I would prefer if we could have a single implementation that is driven by two frontends. The `reqwest` blocking has a reasonable example of unifying the core implementation: https://github.com/seanmonstar/reqwest/blob/master/src/blocking/client.rs

**Pull request closed without merging**